pkg/upgrade/upgrades/v25_2_set_ui_default_timezone.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package upgrades
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/ui"
+	"github.com/cockroachdb/cockroach/pkg/upgrade"
+)
+
+// setUiDefaultTimezone sets the value of the 'ui.default_timezone' cluster
+// setting to be the same as the 'ui.display_timezone' setting if it is set.
+// 'ui.default_timezone' is the new cluster setting to replace
+// 'ui.display_timezone' which doesn't have any restrictions on which timezones
+// can be set.
+func setUiDefaultTimezone(
+	ctx context.Context, version clusterversion.ClusterVersion, d upgrade.TenantDeps,
+) error {
+	// Get the value of the 'ui.display_timezone' setting, if it has been
+	// overridden.
+	row, err := d.DB.Executor().QueryRow(
+		ctx,
+		"get-display-timezone",
+		nil,
+		"SELECT value FROM system.settings WHERE name = 'ui.display_timezone'")
```

Comment:
Just wondering but is it possible to read the setting directly via `d.Settings.SV`?

Reply:
not transactionally with the write of the new setting. That said, the write being done here -- via SET -- isn't transactional either so I guess you could just read the in-mem value and be just as correct?

Reply:
I opted not to use this so that we can distinguish between set and unset. We will only set the new setting if the old setting has explicitly been set. If it is not set (default value is "") it should fall back to UI's defined default tz.

Reply:
if you want to use the cache, you can find out if the value you read is an explicitly set value read from the table vs one that came from the default (or from some external override) by checking `someSetting.ValueOrigin(ctx, &SV) == settings.OriginExplicitlySet`

Reply:
Awesome, thanks for this pointer. Updated to use the value read from `d.Settings.SV`

Reply:
LGTM.

Just to confirm though, as I mentioned earlier, none of this is transactional -- I gathered from the earlier revision here not using a txn that that is okay?

Reply:
Yeah its okay that it is not using a transaction.